### PR TITLE
Feature/update docs

### LINKED ISSUE #449: Add Frontend Style Guide
This should be an ongoing process; we want to ensure that our developers follow the same programming style (which may also be enforced by linters and auto-formatters). This is to  ensure readability and keep a consistent codebase.

### LINKED ISSUE #455: Add Backend Style Guide
This should be an ongoing process; we want to ensure that our developers follow the same programming style (which may also be enforced by linters and auto-formatters). This is to ensure readability and keep a consistent codebase.

### PR DESCRIPTION
**Pull request by JJamesWWang**
### Description
Updates docs to be more relevant to what we use now.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,9 +1 @@
 ### Description
-
```
Comment (yesterd3v) [inline .github/PULL_REQUEST_TEMPLATE.md:2]: Is the aim to remove this just for the `develop` branch? 
On that note, are we now creating a develop and prod branch? I notice some notes on this on JIRA, but I wasn't formally notified on this since I was told before that prod technically is dev.

File: docs/index.rst [hunk 2]
```diff
@@ -19,7 +19,7 @@ Education should be collaborative
 
 Students know best
 ~~~~~~~~~~~~~~~~~~
-    Universities can't keep up with technology. Most university systems aren't even mobile responsive! Forget about using social media. That's why Semester.ly is built by students, and always will be. That's why we're open source! Oh, and it's why we use emojis 🙂.
```
Comment (yesterd3v) [inline docs/index.rst:22]: Why did you remove this line ("That's why we're open source! Oh, and it's why we use emojis 🙂")? I think we should add it back.

File: docs/learning.rst [hunk 3]
```diff
@@ -7,37 +7,59 @@ Learning The Stack
...
 Redux is state container that makes React development easier to manage long term! 
 
-If you're a beginner, `we've created a React tutorial <https://github.com/noahpresler/React-Tutorial>`_ that will teach you the basics of developing a React app all via codepen. This is a great way to get started! 
+The official docs are the go-to: `React Basics
+<https://reactjs.org/docs/hello-world.html>`_, `React Hooks
+<https://reactjs.org/docs/hooks-intro.html>`_, `Redux & Redux Toolkit
+<https://redux.js.org/tutorials/index>`_, and `TypeScript
+<https://www.typescriptlang.org/docs/>`_. We suggest going through the step-by-step
+React tutorial rather than the practical tutorial because the practical tutorial is done
+with class-based components rather than functional components, which we are trying to
```
Comment (yesterd3v) [inline docs/learning.rst]: Remove "which we are trying to shift towards." We can assume they'll notice these changes, and it'll be done fully in the near future.

**Review by yesterd3v — CHANGES_REQUESTED** [review]

File: docs/index.rst (hunk 2 above)
Comment (JJamesWWang) [reply docs/index.rst:22]: Read the commits~~

I said it's because it feels out of place in an otherwise formal pitch

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (JJamesWWang) [reply .github/PULL_REQUEST_TEMPLATE.md:2]: Nah, the PR template is just too annoying

We are shifting from just prod to prod/develop. Mike set us up with all that cuz of how many times I broke staging. Read the contributing section on gitflow for prod/develop 😛

File: docs/learning.rst (hunk 3 above)
Comment (JJamesWWang) [reply docs/learning.rst]: We can't assume this. If 90% of our codebase is class based React, "what's the harm in adding another one?"
"We can just convert it later."
This expectation needs to be clear from the start.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (yesterd3v) [reply .github/PULL_REQUEST_TEMPLATE.md:2]: We can simplify the PR template, but it won't be removed. It's necessary that there's a standard PR description because it's become an issue in the past.

Yeah, that makes sense.

File: docs/index.rst (hunk 2 above)
Comment (yesterd3v) [reply docs/index.rst:22]: If you can note to read commit msgs if they're particularly useful, please do so in the future.

We can leave it out I suppose, but we should maintain some level of informality on the site itself just for future reference.

File: docs/learning.rst (hunk 3 above)
Comment (yesterd3v) [reply docs/learning.rst]: Alright but please add a commented note in the file. I think there's another similar "we're trying to do it now" line elsewhere. The same applies to anything else that'll be removed or changed eventually.

File: docs/index.rst (hunk 2 above)
Comment (JJamesWWang) [reply docs/index.rst:22]: > we should maintain some level of informality on the site itself just for future reference.

What do you mean by this?

File: docs/learning.rst (hunk 3 above)
Comment (JJamesWWang) [reply docs/learning.rst]: I'm not sure what the point is. This is how we end up with so many TODOs. If they're already looking at the .rst file, they were probably looking to update it anyways. I've removed it in favor of "preferring functional components."

File: docs/index.rst (hunk 2 above)
Comment (yesterd3v) [reply docs/index.rst:22]: Typically, it is not expected to read through commit messages and reading code. 
It's more concise if it could be written in brief in the description or the PR description makes a reference to thoroughly read the commit messages.

File: docs/learning.rst (hunk 3 above)
Comment (yesterd3v) [reply docs/learning.rst]: The documentation hasn't been updated in some time, it's helpful for other (and future) developers to do some quick fixes/updates if they know where they are. The hope is that they won't have to read through everything to find something unusual.

**Review by yesterd3v — APPROVED** [review]
LGTM
